### PR TITLE
bpo-1635741: Port faulthandler module to multiphase initialization

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue1635741

### CODE REVIEW

**Review by corona10 — COMMENTED** [review]
@vstinner @shihai1991 PTAL :)

File: Modules/faulthandler.c [hunk 1]
```diff
@@ -1291,59 +1291,56 @@ static PyMethodDef module_methods[] = {
...
-#ifdef MS_WINDOWS
-error:
-    Py_DECREF(m);
-    return NULL;
-#endif
+static struct PyModuleDef module_def = {
+    PyModuleDef_HEAD_INIT,
+    .m_name = "faulthandler",
+    .m_doc = module_doc,
+    .m_size = 0, /* non-negative size to be able to unload the module */
```
Comment (vstinner) [inline Modules/faulthandler.c]: You may remove `.m_size = 0` since it's the default.

**Review by vstinner — APPROVED** [review]
LGTM, but it seems like your PR introduces a compiler warning on the slots:

"initialization from incompatible pointer type [-Wincompatible-pointer-types]"

It may be good to fix it first.

File: Modules/faulthandler.c (hunk 1 above)
Comment (corona10) [reply Modules/faulthandler.c]: Applied!

**Comment by corona10** [comment]
> "initialization from incompatible pointer type [-Wincompatible-pointer-types]"

Yeah, it should be PyModuleDef_Slot, not PyType_Slot, it was my mistake.
I've fixed the PR after the test is passed, I will merge this PR

**Comment by vstinner** [comment]
> Yeah, it should be PyModuleDef_Slot, not PyType_Slot

Ooops :-)

**Comment by vstinner** [comment]
Thanks for taking care of "my" little module :-)

**Pull request merged**